File: app/src/main/java/com/example/josycom/flowoverstack/model/QuestionDataSourceFactory.java
package com.example.josycom.flowoverstack.model;

import androidx.lifecycle.MutableLiveData;
import androidx.paging.DataSource;

public class QuestionDataSourceFactory extends DataSource.Factory<Integer, Question> {

    private final int page;
    private final int pageSize;
    private final String order;
    private final String sortCondition;
    private final String site;
    private final String filter;
    private static MutableLiveData<QuestionDataSource> questionLiveDataSource = new MutableLiveData<>();

    public QuestionDataSourceFactory(int page, int pageSize, String order, String sortCondition, String site, String filter){
        this.page = page;
        this.pageSize = pageSize;
        this.order = order;
        this.sortCondition = sortCondition;
        this.site = site;
        this.filter = filter;
    }

    @Override
    public DataSource<Integer, Question> create() {
        QuestionDataSource questionDataSource = new QuestionDataSource(page, pageSize, order, sortCondition, site, filter);
        questionLiveDataSource.postValue(questionDataSource);
        return questionDataSource;
    }

    public static MutableLiveData<QuestionDataSource> getQuestionLiveDataSource() {
        return questionLiveDataSource;
    }
}

File: app/src/main/java/com/josycom/mayorjay/flowoverstack/data/QuestionDataSourceFactory.kt
package com.josycom.mayorjay.flowoverstack.data

import androidx.lifecycle.MutableLiveData
import androidx.paging.DataSource
import com.josycom.mayorjay.flowoverstack.model.Question
import com.josycom.mayorjay.flowoverstack.network.ApiService

class QuestionDataSourceFactory(private val page: Int, private val pageSize: Int, private val order: String,
                                private val sortCondition: String, private val site: String, private val tagged: String, private val filter: String,
                                private val siteKey: String, private val apiService: ApiService) : DataSource.Factory<Int, Question>() {

    override fun create(): DataSource<Int, Question> {
        val questionDataSource = QuestionDataSource(page, pageSize, order, sortCondition, site, tagged, filter, siteKey, apiService)
        questionLiveDataSource.postValue(questionDataSource)
        return questionDataSource
    }

    companion object {
        val questionLiveDataSource = MutableLiveData<QuestionDataSource>()
    }
}